EncodingStreamingHighAvailability/HighAvailability.Provisioning/ProvisioningFunction.cs

```diff
@@ -0,0 +1,57 @@
+namespace HighAvailabikity.Provisioner
+{
+    using HighAvailability.Interfaces;
+    using HighAvailability.Models;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Implements provisioning Azure function. It is triggered by messages in provisioning-requests Azure queue
+    /// </summary>
+    public class ProvisioningFunction
+    {
+        /// <summary>
+        /// Service to orchestrate provisioning
+        /// </summary>
+        private IProvisioningOrchestrator provisioningOrchestrator { get; set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="provisioningOrchestrator">Service to orchestrate provisioning</param>
+        public ProvisioningFunction(IProvisioningOrchestrator provisioningOrchestrator)
+        {
+            this.provisioningOrchestrator = provisioningOrchestrator ?? throw new ArgumentNullException(nameof(provisioningOrchestrator));
+        }
+
+        /// <summary>
+        /// This function is triggered by messages in provisioning-requests Azure queue.
+        /// It runs provisioning logic for encoded assets.
```

Comment:
Not all jobs are encoding.   Should say "processed"

Reply:
fixed